molecule/default/verify.yml

```diff
@@ -9,30 +9,27 @@
   tasks:
 
     - name: Gather EC2 facts
+      delegate_to: "{{ groups['nginx'][0] }}"
```

Comment:
I think this is useless as fact are only ever delegated if delegate_facts is set to true.
Also in the later tasks the ec2 facts you used are used from localhost (not delegated) tasks.